From b7f727ac83ae92db61dddf414822b05eee0f2b69 Mon Sep 17 00:00:00 2001
From: Stephan Mueller <smueller@chronox.de>
Date: Mon, 30 May 2022 21:55:07 +0200
Subject: [PATCH 2/2] lrng: add random_init function

With 5.18.1, the existing function of rand_initialize was replaced with
random_init. It has the same purpose but provides the kernel command
line string that can be used to stir the auxiliary pool.

Signed-off-by: Stephan Mueller <smueller@chronox.de>
---
 lrng_interface_random_kernel.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/char/lrng/lrng_interface_random_kernel.c b/drivers/char/lrng/lrng_interface_random_kernel.c
index 4dae08e06500..81060ba98742 100644
--- a/drivers/char/lrng/lrng_interface_random_kernel.c
+++ b/drivers/char/lrng/lrng_interface_random_kernel.c
@@ -25,9 +25,12 @@ static unsigned int lrng_ready_chain_used = 0;
 
 /********************************** Helper ***********************************/
 
-int __init rand_initialize(void)
+int __init random_init(const char *command_line)
 {
-	return lrng_rand_initialize();
+	int ret = lrng_rand_initialize();
+
+	lrng_pool_insert_aux(command_line, strlen(command_line), 0);
+	return ret;
 }
 
 bool lrng_ready_chain_has_sleeper(void)
-- 
2.36.1

